daemon/cmd/kube_proxy_replacement.go

```diff
@@ -272,6 +272,24 @@ func initKubeProxyReplacementOptions() (strict bool) {
 		}
 	}
 
+	if option.Config.SkipSocketLBInPodNamespace {
+		if !option.Config.EnableHostReachableServices {
+			log.Warnf("SkipSocketLBInPodNamespace only takes effect when EnableHostReachableServices is true")
+		}
+		found := false
+		if helpers := probesManager.GetHelpers("cgroup_sock_addr"); helpers != nil {
+			if _, ok := helpers["bpf_get_netns_cookie"]; ok {
+				found = true
+			}
+		}
+		if !found {
+			log.Warn("Without network namespace cookie lookup functionality, BPF datapath " +
+				"cannot distinguish root and non-root namespace, skipping socket-level " +
+				"loadbalancing will not work. Istio routing chains will be missed. " +
+				"Needs kernel version >= 5.7")
```

Comment:
shouldn't we then also set option.Config.SkipSocketLBInPodNamespace to false?

Reply:
It doesn't really do anything without the helper, but I agree, setting it to false explicitly here is better. Done.